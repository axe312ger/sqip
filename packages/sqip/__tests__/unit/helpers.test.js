import { readFileSync } from 'fs'
import { resolve } from 'path'

import {
  encodeBase64,
  getDimensions,
  printFinalResult
} from '../../src/helpers'

const rawSVG = readFileSync(
  resolve(__dirname, '../../../../demo/beach-sqip.svg')
)

const logSpy = jest.spyOn(global.console, 'log')

test('encodeBase64', () => {
  const result = encodeBase64(rawSVG)
  expect(result).toMatchSnapshot()
})

test('getDimensions', () => {
  const result = getDimensions(resolve(__dirname, '../../../../demo/beach.jpg'))
  expect(result).toMatchSnapshot()
})

test('printFinalResult', () => {
  const width = 200
  const height = 100
  printFinalResult(
    { width, height },
    '/path/to/image.jpg',
    'data:image/svg+xml;base64,base64EncodedSVG'
  )
  expect(logSpy).toHaveBeenCalledWith(
    '<img width="200" height="100" src="/path/to/image.jpg" alt="Add descriptive alt text" style="background-size: cover; background-image: url(data:image/svg+xml;base64,base64EncodedSVG);">'
  )
})
