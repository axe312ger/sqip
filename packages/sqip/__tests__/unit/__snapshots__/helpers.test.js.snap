// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`encodeBase64 1`] = `"PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxMDI0IDY0MCI+PGZpbHRlciBpZD0iYiI+PGZlR2F1c3NpYW5CbHVyIHN0ZERldmlhdGlvbj0iMTIiIC8+PC9maWx0ZXI+PHBhdGggZmlsbD0iIzcxOGQ5NSIgZD0iTTAgMGgxMDI0djY0MEgweiIvPjxnIGZpbHRlcj0idXJsKCNiKSIgdHJhbnNmb3JtPSJtYXRyaXgoNCAwIDAgNCAyIDIpIiBmaWxsLW9wYWNpdHk9Ii41Ij48ZWxsaXBzZSBmaWxsPSIjZmZiNTM3IiByeD0iMSIgcnk9IjEiIHRyYW5zZm9ybT0ibWF0cml4KC0uNjk0NzcgMTUuMDkzMjYgLTczLjYzMzI4IC0zLjM4OTUgMTUzLjMgMjkuOCkiLz48ZWxsaXBzZSBmaWxsPSIjNDUxYjA4IiBjeD0iMTk0IiBjeT0iMTU5IiByeD0iNDIiIHJ5PSI3MSIvPjxwYXRoIGZpbGw9IiNmZmZkY2UiIGQ9Ik0xNDQuMyAxNzguMmwtMTQ0IDIuNi0uNi00MSAxNDQtMi42eiIvPjxlbGxpcHNlIGZpbGw9IiMyNTdhOTQiIHJ4PSIxIiByeT0iMSIgdHJhbnNmb3JtPSJtYXRyaXgoLjMwMTMzIDI2LjUzNTk5IC0xMDIuMTU4MzYgMS4xNjAwNyA3OS40IDEwNykiLz48ZWxsaXBzZSBmaWxsPSIjNzJhY2Q1IiByeD0iMSIgcnk9IjEiIHRyYW5zZm9ybT0ibWF0cml4KC03LjIzMTU2IDI1LjU1NTc5IC0xMTMuMDYwMDYgLTMxLjk5Mjc3IDMyLjYgNDIuMSkiLz48ZWxsaXBzZSBmaWxsPSIjMzQ4YmEyIiByeD0iMSIgcnk9IjEiIHRyYW5zZm9ybT0ibWF0cml4KC0zMi4wMjEwNyAyMi4zOTE5OCAtMTQuNjAwODIgLTIwLjg3OTUzIDIzNS44IDExMC40KSIvPjxlbGxpcHNlIGZpbGw9IiNmZmE4NjYiIHJ4PSIxIiByeT0iMSIgdHJhbnNmb3JtPSJtYXRyaXgoMS40Mjg0MiAxMy45Mzk1NCAtNTkuNDQ5NjggNi4wOTE5NiAxMzIgMjgpIi8+PC9nPjwvc3ZnPgo="`;

exports[`getDimensions 1`] = `
Object {
  "height": 640,
  "type": "jpg",
  "width": 1024,
}
`;
